src/overlays/actors/ovl_Bg_Fire_Wall/z_bg_fire_wall.c

```diff
@@ -32,41 +33,184 @@ const ActorInit Bg_Fire_Wall_InitVars = {
+        this->timer = 5;
+        this->actionFunc = func_809AC68C;
+    }
+}
+
+void func_809AC68C(BgFireWall* this, PlayState* play) {
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    if (this->timer == 0) {
+        this->actionFunc = func_809AC6C0;
+    }
+}
+
+void func_809AC6C0(BgFireWall* this, PlayState* play) {
+    if ((this->unk_14C != 0) || (func_809AC5C0(this, play))) {
+        this->unk_15C = 0.09f;
+        Math_StepToF(&this->actor.scale.y, this->unk_15C + this->unk_154, this->unk_158);
+    } else if (Math_StepToF(&this->actor.scale.y, 0.005f, this->unk_158)) {
+        this->actionFunc = func_809AC638;
+    } else {
+        this->timer = 0;
+    }
+}
+
+void func_809AC760(BgFireWall* this, PlayState* play) {
+    s16 phi_a3;
+
+    if (Actor_IsFacingPlayer(&this->actor, 0x4000)) {
+        phi_a3 = this->actor.shape.rot.y;
+    } else {
+        phi_a3 = (this->actor.shape.rot.y + 0x8000);
+    }
+    func_800B8D98(play, &this->actor, BREG(48) + 10.0f, phi_a3, BREG(49) + 5.0f);
+}
+
+void func_809AC7F8(BgFireWall* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    Vec3f sp38;
+    f32 val = 25.0f;
+    f32 sin;
+    f32 cos;
+
+    Actor_OffsetOfPointInActorCoords(&this->actor, &sp38, &player->actor.world.pos);
+    sp38.x = CLAMP(sp38.x, -80.0f, 80.0f);
+
+    if (this->step == 0) {
+        if (sp38.z > 0.0f) {
+            sp38.z = -val;
+            this->step = -1.0f;
+        } else {
+            sp38.z = val;
+            this->step = 1.0f;
+        }
+    } else {
+        sp38.z = this->step * 25.0f;
+    }
+
+    sin = Math_SinS(this->actor.shape.rot.y);
+    cos = Math_CosS(this->actor.shape.rot.y);
+    this->collider.dim.pos.x = this->actor.world.pos.x + (sp38.x * cos) + (sp38.z * sin);
+    this->collider.dim.pos.z = this->actor.world.pos.z - (sp38.x * sin) + (sp38.z * cos);
+}
+
+void func_809AC970(BgFireWall* this, PlayState* play) {
+    if (Math_StepToF(&this->actor.scale.y, 0.005f, this->unk_158)) {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void BgFireWall_Update(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    BgFireWall* this = THIS;
+
+    this->actionFunc(this, play);
+    if ((this->unk_14C == 0) || ((this->unk_14C != 0) && (this->actor.xzDistToPlayer < 240.0f))) {
+        if (this->collider.base.atFlags & AT_HIT) {
+            this->collider.base.atFlags &= ~AT_HIT;
+            func_809AC760(this, play);
+        }
+    }
+    if (this->actionFunc == func_809AC6C0) {
+        func_800B9010(&this->actor, NA_SE_EV_FIRE_PLATE - SFX_FLAG);
+        if ((this->unk_14C == 0) || ((this->unk_14C != 0) && (this->actor.xzDistToPlayer < 240.0f))) {
+            func_809AC7F8(this, play);
+            CollisionCheck_SetAT(play, &play->colChkCtx, &this->collider.base);
+            CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+        }
+    }
+    this->texIndex = (this->texIndex + 1) % 8;
```

Comment:
This is incrementing the index up to the size of the array, so I think using an ARRAY_COUNT here would be a bit more appropriate 
```suggestion
    this->texIndex = (this->texIndex + 1) % ARRAY_COUNT(sFlameTextures);
```